### PR TITLE
Site Next App: Add SitePreviewProvider in draftMode/site preview

### PR DESCRIPTION
**Pull request by nsams**

### CODE REVIEW

File: packages/site/cms-site/src/sitePreview/SitePreviewProvider.tsx [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 import { sendSitePreviewIFrameMessage } from "./iframebridge/sendSitePreviewIFrameMessage";
 import { SitePreviewIFrameLocationMessage, SitePreviewIFrameMessageType } from "./iframebridge/SitePreviewIFrameMessage";
 
 const SitePreview: React.FunctionComponent<{ children: React.ReactNode }> = ({ children }) => {
-    const router = useRouter();
+    const pathname = usePathname();
 
     React.useEffect(() => {
         function sendUpstreamMessage() {
-            const url = new URL(router.asPath, window.location.origin);
```
Comment (nsams) [inline packages/site/cms-site/src/sitePreview/SitePreviewProvider.tsx:13]: useRouter.asPath did include search params, usePathname doesn't. And next h as no api to get search, that's why I added location.search for now.

This doesn't update if search changes!

File: demo/site/src/app/layout.tsx [hunk 2]
```diff
@@ -13,10 +16,13 @@ export default function RootLayout({
 }: Readonly<{
     children: React.ReactNode;
 }>) {
+    const SitePreviewWrapper = draftMode().isEnabled ? SitePreviewProvider : Fragment;
```
Comment (johnnyomair) [inline demo/site/src/app/layout.tsx:19]: Do you think it's better to do this here than in SitePreviewProvider?

File: packages/site/cms-site/src/sitePreview/SitePreviewProvider.tsx (hunk 1 above)
Comment (johnnyomair) [reply packages/site/cms-site/src/sitePreview/SitePreviewProvider.tsx:13]: How about [useSearchParams](https://nextjs.org/docs/app/api-reference/functions/use-search-params)?

Comment (nsams) [reply packages/site/cms-site/src/sitePreview/SitePreviewProvider.tsx:13]: ah, yes of course :) sorry

File: demo/site/src/app/layout.tsx (hunk 2 above)
Comment (nsams) [reply demo/site/src/app/layout.tsx:19]: SitePreviewProvider is (must be) a client component, draftMode() is for server only.

But I could split SitePreviewProvider into to files.

I think it's better that way, as it avoids having a "SitePreviewProvider" wrapper in non-preview/draft mode. Even if it does nothing, it's kind of confusing to have it.

Before app-router we had a custom file for preview paths and only there the SitePreviewProvider was wrapped.

Comment (johnnyomair) [reply demo/site/src/app/layout.tsx:19]: > Before app-router we had a custom file for preview paths and only there the SitePreviewProvider was wrapped.

@fraxachun changed that in [#1237](https://github.com/vivid-planet/comet/pull/1237/files#diff-5ae39b382bf2224df93c5f3674fdb1a6c2373ef44372df055c24e8aa2aa4a77a).

Comment (johnnyomair) [reply demo/site/src/app/layout.tsx:19]: I'm not sure about the `SitePreviewWrapper` name though. Maybe `ConditionalSitePreview`? Or do it inline in the JSX?

Comment (nsams) [reply demo/site/src/app/layout.tsx:19]: > Or do it inline in the JSX?

how?

Comment (johnnyomair) [reply demo/site/src/app/layout.tsx:19]: > how?

```tsx
<StyledComponentsRegistry>
  {draftMode().isEnabled ? <SitePreviewProvider>{children}</SitePreviewProvider> : children}
</StyledComponentsRegistry>
```

But maybe that would get more complicated if children become more complex.

Comment (nsams) [reply demo/site/src/app/layout.tsx]: done

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**